picard/tagger.py

```diff
@@ -485,21 +485,26 @@ def _file_loaded(self, file, target=None, remove_file=False, unmatched_files=Non
             self.cluster(unmatched_files)
 
     def move_file(self, file, target):
-        if target is None:
-            if not file.parent:
-                target = self.unclustered_files
-            else:
-                log.debug("Aborting move since target is invalid and file already has a parent")
+        """Moves a file to target, if possible
+
+        Returns the actual target the files has been moved to or None
+        """
         if isinstance(target, Album):
             self.move_files_to_album([file], album=target)
-        elif isinstance(target, File) and target.parent:
-            file.move(target.parent)
-        # To be able to move a file to it target must implement add_file(file)
-        elif hasattr(target, 'add_file'):
-            file.move(target)
-        # Ensure a file always has a parent so it shows up in UI
-        elif not file.parent:
-            file.move(self.unclustered_files)
+        else:
+            if isinstance(target, File) and target.parent:
+                target = target.parent
+            # To be able to move a file the target must implement add_file(file)
+            if not hasattr(target, 'add_file'):
+                # Ensure a file always has a parent so it shows up in UI
+                if not file.parent:
+                    target = self.unclustered_files
+                # Unsupported target, do not move the file
+                else:
+                    target = None
+            if target is not None:
```

Comment:
I think `File.move()` should handled the `target is None` case itself.

https://github.com/metabrainz/picard/blob/1415787a873e71667b605a68afd00305d2d60417/picard/file.py#L611-L623

If we call `File.move(None)`, it will likely raise an exception when hitting https://github.com/metabrainz/picard/blob/1415787a873e71667b605a68afd00305d2d60417/picard/file.py#L618 or https://github.com/metabrainz/picard/blob/1415787a873e71667b605a68afd00305d2d60417/picard/file.py#L622

So I think we should check for it:

```python
    def move(self, parent):
        if parent is not None and parent != self.parent:
```

Reply:
Good point. Or actually even do check for `hasattr(parent, 'add_file')`

Reply:
@zas Please check 598417a22a940b8a7104eb92339575c111f60eef